.github/workflows/main.yaml

```diff
@@ -173,7 +172,7 @@ jobs:
       - name: Test
         run: |
           make test
-        if: matrix.target != 'aarch64-apple-darwin'
+        if: matrix.build != 'macos-arm64'
```

Comment:
Are target and build the same here? It seems like they could be different

Reply:
`matrix.target` and `matrix.os` are names defined by either Rust or GitHub. `matrix.build` is something we chosse on our side so I prefer to rely on our naming convention here. 

If GitHub forces us to change the `os` this will not impact the workflow. Same for `target` (even if it's unlikely they will change)

Reply:
Sure that makes sense, I just want to check that there aren't extra semantics here. If I remember correctly, arm64 Mac used `target` or some other key for a very specific purpose and I think the difference might have been important because of cross-compilation or something

Reply:
@markandrus yeah, to build for M1 we cross compile so we define (in the matrix) the Rust target to use and we were also using this in our conditionals.

But now, we have a problem: when GitHub will support M1 directly, we won't cross-compile anymore (we won't force the target through the matrix). Without my change, this would also require to change all the conditionals. With my change, we only update the matrix and we are done.